Server/models/user.js

```diff
@@ -0,0 +1,45 @@
+const mongoose = require('mongoose')
+const bcrypt = require('bcrypt')
+
+const UserSchema = mongoose.Schema({
+    username: {
+        type: String
+    },
+    email: {
+        type: String,
+        required: true
+    },
+    password: {
+        type: String,
+        required: true
+    },
+    profilePicUrl: {
+        type: String
+    },
+    isActive: {
+        type: Boolean,
+        default: true
+    },
+    isVerified: {
+        type: Boolean,
+        default: false
+    },
+    updated_at: {
+        type: Date,
+        default: Date.now
+    },
+    created_at: {
+        type: Date,
+        default: Date.now
+    }
+})
+
+UserSchema.pre('save', async function (next) {
```

Comment:
This is cool, I did not know about the `pre` function

Reply:
Yeah It's pretty useful.